plugins/systemd/systemd_units

```diff
@@ -123,8 +125,12 @@ fetch () {
 		count=$(echo "$state_unit_list" | grep -c "$state$")
 		echo "$state.value $count"
 		extinfo=$(echo "$state_unit_list" | grep "$state$" | cut -d " " -f 1 | tr '\n' ' ')
-		if [ -n "$extinfo" ]; then
-			echo "$state.extinfo" "$extinfo"
+		if [ -z "$silence_active_extinfo" ]; then
+			[ -n "$extinfo" ] && echo "$state.extinfo" "$extinfo"
```

Comment:
This short conditional expression (`... && ...`) usually works fine, but should be avoided at the end of complicated expressions (here `if ... fi`). Otherwise the non-zero exitcode (if `extinfo` is empty) would traverse up to the higher layers and maybe even end up as an unexpected exit code of the plugin.

This issue is more relevant for scripts using the *exit on error* setting (`set -e`), but it is applicable here, too.

Reply:
Fixed.